compiler/src/dotty/tools/dotc/transform/InterceptedMethods.scala

```diff
@@ -105,7 +105,7 @@ class InterceptedMethods extends MiniPhase {
               List(qual, typer.resolveClassTag(tree.pos, qual.tpe.widen))))
           }*/
          */
-      case t if primitiveGetClassMethods.contains(t) =>
+      case t if t.name == nme.getClass_ && scalaValueClasses.contains(t.owner) =>
```

Comment:
I think you could just do defn.ScalaValueClasses.contains, no need for the intermediate variable.